src/main/java/com/commercetools/sync/customobjects/CustomObjectSync.java

```diff
@@ -0,0 +1,329 @@
+        @Nonnull final CustomObjectDraft<JsonNode> customObjectDraft) {
+
+        return syncOptions
+            .applyBeforeCreateCallback(customObjectDraft)
+            .map(draft -> customObjectService.upsertCustomObject(customObjectDraft)
+                                             .thenApply(customObjectOptional -> {
+                                                 if (customObjectOptional.isPresent()) {
+                                                     statistics.incrementCreated();
+                                                 } else {
+                                                     statistics.incrementFailed();
+                                                 }
+                                                 return customObjectOptional;
+                                             })
+            )
+            .orElse(CompletableFuture.completedFuture(Optional.empty()));
+    }
+
+
+    /**
+     * Given an existing {@link CustomObject} and a new {@link CustomObjectDraft}, the method verifies whether the JSON
+     * objects between {@link CustomObject#getValue()} and {@link CustomObjectDraft#getValue()} are identical. If so, a
+     * request is made to CTP to update the existing custom object, otherwise it doesn't issue a request.
+     *
+     * <p>The {@code statistics} instance is updated accordingly to whether the CTP request was carried
+     * out successfully or not. If an exception was thrown on executing the request to CTP,the error handling method
+     * is called.
+     *
+     * @param oldCustomObject existing custom object that could be updated.
+     * @param newCustomObject draft containing data that could differ from data in {@code oldCustomObject}.
+     * @return a {@link CompletionStage} which contains an empty result after execution of the update.
+     */
+    @Nonnull
+    private CompletionStage<Optional<CustomObject<JsonNode>>> updateCustomObject(
+        @Nonnull final CustomObject<JsonNode> oldCustomObject,
+        @Nonnull final CustomObjectDraft<JsonNode> newCustomObject) {
+
+        if (CustomObjectSyncUtils.hasIdenticalValue(oldCustomObject, newCustomObject)) {
+            return customObjectService
+                .upsertCustomObject(newCustomObject)
+                .handle(ImmutablePair::new)
+                .thenCompose(updateResponse -> {
+                    final CustomObject<JsonNode> updatedCustomObject = updateResponse.getKey().get();
+                    final Throwable sphereException = updateResponse.getValue();
+                    if (sphereException != null) {
+                        return executeSupplierIfConcurrentModificationException(sphereException,
+                            () -> fetchAndUpdate(oldCustomObject, newCustomObject),
+                            () -> {
+                                final String errorMessage =
+                                    format(CTP_CUSTOM_OBJECT_UPSERT_FAILED,
```

Comment:
You know this is an update not a create request so the error message should be clear.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/569/commits/893a6d19adde88df3b3d5a973800c0bf0cb5dadc